src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking_FlowAnalysis.cs

```diff
@@ -2875,5 +3025,179 @@ static object G(C c, I<string?> x, I<object?> y)
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "x").WithArguments("IB<object?>", "IA<object>").WithLocation(13, 13),
+                // (17,25): warning CS8619: Nullability of reference types in value of type 'IB<object>' doesn't match target type 'IA<object?>'.
+                //         IA<object?> y = x;
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "x").WithArguments("IB<object>", "IA<object?>").WithLocation(17, 25),
+                // (18,13): warning CS8619: Nullability of reference types in value of type 'IB<object>' doesn't match target type 'IA<object?>'.
+                //         y = x;
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "x").WithArguments("IB<object>", "IA<object?>").WithLocation(18, 13));
+        }
+
+        [Fact]
+        public void Conversions_03()
+        {
+            var source =
+@"interface IOut<out T> { }
+class C
+{
+    static void F(IOut<object> x)
+    {
+        IOut<object?> y = x;
+    }
+    static void G(IOut<object?> x)
+    {
+        IOut<object> y = x;
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics(
+                // (10,26): warning CS8619: Nullability of reference types in value of type 'IOut<object?>' doesn't match target type 'IOut<object>'.
+                //         IOut<object> y = x;
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "x").WithArguments("IOut<object?>", "IOut<object>").WithLocation(10, 26));
+        }
+
+        [Fact]
+        public void Conversions_04()
+        {
+            var source =
+@"interface IIn<in T> { }
+class C
+{
+    static void F(IIn<object> x)
+    {
+        IIn<object?> y = x;
+    }
+    static void G(IIn<object?> x)
+    {
+        IIn<object> y = x;
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics(
+                // (6,26): warning CS8619: Nullability of reference types in value of type 'IIn<object>' doesn't match target type 'IIn<object?>'.
+                //         IIn<object?> y = x;
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "x").WithArguments("IIn<object>", "IIn<object?>").WithLocation(6, 26));
+        }
+
+        [Fact]
+        public void Conversions_05()
+        {
+            var source =
+@"interface IOut<out T> { }
+class A<T> : IOut<T> { }
+class C
+{
+    static void F(A<string> x)
+    {
+        IOut<object?> y = x;
+    }
+    static void G(A<string?> x)
+    {
+        IOut<object> y = x;
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics(
+                // (11,26): warning CS8619: Nullability of reference types in value of type 'A<string?>' doesn't match target type 'IOut<object>'.
```

Comment:
>// (11,26): warning CS8619: Nullability of reference types in value of type 'A<string?>' doesn't match target type 'IOut<object>'. [](start = 16, length = 130)

It looks like warnings like this can be very confusing, ```A``` might be non generic, or its type arguments might have nothing to do with the mismatch. Consider adding tests for scenarios like that.

Reply:
As you mentioned, `Conversions_06` is an example.

---
In reply to: [183875742](https://github.com/dotnet/roslyn/pull/26356#discussion_r183875742) [](ancestors = 183875742)